pkg/reconciler/v1alpha1/revision/resources/deploy.go

```diff
@@ -114,7 +115,10 @@ func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, observab
 	// Adding or removing an overwritten corev1.Container field here? Don't forget to
 	// update the validations in pkg/webhook.validateContainer.
 	userContainer.Name = UserContainerName
-	userContainer.Resources = userResources
+
+	if equality.Semantic.DeepEqual(userContainer.Resources, corev1.ResourceRequirements{}) {
+		userContainer.Resources = userResources
```

Comment:
I wonder if we should do a deep merge on CPU here? IIUC if I specify only a memory resource here I will get an implicit undefinition of CPU resources.

Reply:
That makes lot of sense. I thought about it as well.